### DISCUSSION TITLE
Open source on-premise alternative

### DISCUSSION THREAD

**Discussion by its-engineersindia** [mayooear/ai-pdf-chatbot-langchain#78] [Q&A]
Can someone suggest open-source and on-premise deployable alternatives for GPT4, Pinecone, etc. to achieve similar results?

**Comment by alew3** [comment]
Check Alpaca https://medium.com/@martin-thissen/llama-alpaca-chatgpt-on-your-local-computer-tutorial-17adda704c23

**Comment by juliensalinas** [comment]
You can deploy cutting edge open-source models like LLaMA 2 or Mistral 7B by yourself. Here is a quick tuto about deploying Mistral 7B: https://nlpcloud.com/deploy-mistral-7b-on-a10-gpu-on-aws.html

Another approach is to leverage NLP Cloud's on-premise offer (in that case this is not open source but you can download NLP Cloud's generative models on your own servers).
